cmd/config-migrate.go

```diff
@@ -1110,3 +1114,145 @@ func migrateV14ToV15() error {
 	)
 	return nil
 }
+
+// Version '15' to '16' migration. Adds "format" configuration
+// parameter for database targets.
+func migrateV15ToV16() error {
+	cv15, err := loadConfigV15()
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return fmt.Errorf("Unable to load config version ‘14’. %v", err)
+	}
+	if cv15.Version != "15" {
+		return nil
+	}
+
+	// Copy over fields from V15 into V16 config struct
+	srvConfig := &serverConfigV16{
+		Logger: &logger{},
+		Notify: &notifier{},
+	}
+	srvConfig.Version = "16"
+	srvConfig.Credential = cv15.Credential
+	srvConfig.Region = cv15.Region
+	if srvConfig.Region == "" {
+		// Region needs to be set for AWS Signature Version 4.
+		srvConfig.Region = globalMinioDefaultRegion
+	}
+	srvConfig.Logger.Console = cv15.Logger.Console
+	srvConfig.Logger.File = cv15.Logger.File
+
+	// check and set notifiers config
+	if len(cv15.Notify.AMQP) == 0 {
+		srvConfig.Notify.AMQP = make(map[string]amqpNotify)
+		srvConfig.Notify.AMQP["1"] = amqpNotify{}
+	} else {
+		srvConfig.Notify.AMQP = cv15.Notify.AMQP
+	}
+	if len(cv15.Notify.ElasticSearch) == 0 {
+		srvConfig.Notify.ElasticSearch = make(map[string]elasticSearchNotify)
+		srvConfig.Notify.ElasticSearch["1"] = elasticSearchNotify{}
+	} else {
+		// IMPORTANT NOTE: Future migrations should remove
+		// this as existing configuration will already contain
+		// a value for the "format" parameter.
+		for k, v := range cv15.Notify.ElasticSearch.Clone() {
+			if v.Enable {
```

Comment:
We are adding a parameter called "format" to 4 targets (postgres, mysql, elasticsearch and redis). The values can be "namespace" (existing behaviour) and "access" (new behaviour where all events are appended in the dbs). For existing user configurations, the value has to be "namespace", so that is what the migration in this commit is doing. There is no default, as the user is expected to choose the kind of behaviour they want, and the parameter is ignored if Enable=false. For future migrations (say version 16 to 17), we no longer need to migrate for "format" values and the indicated loop above should be removed - we should just be copying over existing format values.